### PR TITLE
Add auth CLI flag

### PR DESCRIPTION
**Pull request by paulgb**

### CODE REVIEW

File: src/jamsocket.ts [hunk 1]
```diff
@@ -44,16 +44,19 @@ export class Jamsocket {
...
 
     const body: SpawnRequestBody = {
       env,
       grace_period_seconds: grace,
       port,
       tag,
+      require_bearer_token: requireBearerToken,
     }
 
+    console.log('body', body)
```
Comment (rolyatmax) [inline src/jamsocket.ts]: Can drop this?

File: src/commands/spawn.ts [hunk 2]
```diff
@@ -20,6 +20,7 @@ export default class Spawn extends Command {
     grace: Flags.integer({ char: 'g', description: 'optional grace period (in seconds) to wait after last connection is closed before shutting down container' }),
     port: Flags.integer({ char: 'p', description: 'optional port for jamsocket to proxy requests to (default is 8080)', hidden: true }),
     tag: Flags.string({ char: 't', description: 'optional tag for the service to spawn (default is latest)' }),
+    requireBearerToken: Flags.boolean({ char: 'r', description: 'require a bearer token to access the service' }),
```
Comment (rolyatmax) [inline src/commands/spawn.ts]: Does this mean the CLI flag is `--requireBearerToken` or `--require-bearer-token`?

Also, maybe the description is something like "generate a bearer token for accessing the backend"?

File: src/jamsocket.ts (hunk 1 above)
Comment (paulgb) [reply src/jamsocket.ts]: oops yeah

File: src/commands/spawn.ts (hunk 2 above)
Comment (paulgb) [reply src/commands/spawn.ts]: Ah, I didn't realize oclif exposed the name. I've changed it to `--require-bearer-token` and added to the description.

**Review by rolyatmax — APPROVED** [review]
LGTM!

**Comment by rolyatmax** [comment]
I'll cut v0.4.8!

**Comment by paulgb** [comment]
> 

Thanks!

**Pull request merged**